src/components/forms/FormWorkflow.tsx

```diff
@@ -0,0 +1,150 @@
+import React from "react";
+import type { SyntheticEvent, Dispatch } from "react";
+import { Button, Form, Stepper, useToggle } from "@edx/paragon";
+
+import ConfigError from "../settings/ConfigError";
+// @ts-ignore
+import { useFormContext } from "./FormContext.tsx";
+import type {
+  FormFields,
+  FormFieldValidation,
+  FormContext,
+} from "./FormContext";
+// @ts-ignore
+import { setStepAction } from "./data/actions.ts";
+import { SUBMIT_TOAST_MESSAGE } from "../settings/data/constants";
+import { FormActionArguments } from "./data/actions";
+// @ts-ignore
+import UnsavedChangesModal from "../settings/SettingsLMSTab/UnsavedChangesModal.tsx";
+
+export type FormWorkflowButtonConfig<FormData> = {
+  buttonText: string;
+  onClick: (formFields: FormData) => Promise<void>;
+};
+
+type DynamicComponent = React.FunctionComponent | React.ComponentClass;
+
+export type FormWorkflowStep<FormData> = {
+  index: number;
+  stepName: string;
+  formComponent: DynamicComponent;
+  validations: FormFieldValidation[];
+  saveChanges: (FormData) => Promise<void>;
+  nextButtonConfig: FormWorkflowButtonConfig<FormData>;
+};
+
+export type FormWorkflowConfig<FormData> = {
+  steps: FormWorkflowStep<FormData>[];
+  getCurrentStep: () => FormWorkflowStep<FormData>;
+};
+
+export type FormWorkflowProps<FormData> = {
+  formWorkflowConfig: FormWorkflowConfig<FormData>;
+  onClickOut: (edited: boolean, msg?: string) => null;
+  formData: FormData;
+  dispatch: Dispatch<FormActionArguments>;
+};
+
+// Modal container for multi-step forms
+function FormWorkflow<FormData>({
+  formWorkflowConfig,
+  onClickOut,
+  dispatch,
+}: FormWorkflowProps<FormData>) {
+  const {
+    formFields,
+    currentStep: step,
+    hasErrors,
+    isEdited,
+  }: FormContext = useFormContext();
+  const [errorIsOpen, openError, closeError] = useToggle(false);
+  const [savedChangesModalIsOpen, openModal, closeModal] = useToggle(false);
```

Comment:
nit: I think the open and closeModal vars should be better named like openSaveModal, closeSaveModal